eve/methods/delete.py

```diff
@@ -193,20 +193,34 @@ def delete(resource, **lookup):
 
     .. versionadded:: 0.0.2
     """
-    getattr(app, "on_delete_resource")(resource)
-    getattr(app, "on_delete_resource_%s" % resource)()
 
     resource_def = config.DOMAIN[resource]
+    getattr(app, "on_delete_resource")(resource)
+    getattr(app, "on_delete_resource_%s" % resource)()
+    default_request = ParsedRequest()
+    if config.DOMAIN[resource]['soft_delete']:
+        # get_document should always fetch soft deleted documents from the db
+        # callers must handle soft deleted documents
+        default_request.show_deleted = True
```

Comment:
Well, in this case, I would like to offer to the client possibility to customize what he wants to delete. A possible use case can be that in the collection there are documents which should not be deleted by the client (i.e. included service for a trip, like included car or included checked baggage).

Therefore, it might be possible by modifying the query (in the case on hard delete) or modify the originals (removing what we do not what to delete) in the callback.